src/pudl/analysis/imputation/timeseries_cleaning.py

```diff
@@ -0,0 +1,1654 @@
+            pd.DataFrame({"id": mask.columns[col], "year": mask.index[row]})
+            .groupby("id")["year"]
+            .apply(lambda x: np.sort(x))
+        )
+        with pd.option_context("display.max_colwidth", None):
+            logger.info(f"{msg}:\n{report}")
+    # Drop respondents with no data
+    blank = df.columns[df.isnull().all()].tolist()
+    df = df.drop(columns=blank)
+    # Report dropped respondents (with no data)
+    logger.info(f"Dropped blank respondents: {blank}")
+    return df
+
+
+def impute_timeseries_asset_factory(
+    input_asset_name: str,
+    output_asset_name: str,
+    years: list[int],
+    value_col: str = "demand_mwh",
+    imputed_value_col: str = "demand_imputed_mwh",
+    reported_value_col: str = "demand_reported_mwh",
+    id_col: str = "",
+    output_io_manager_key: str = "parquet_io_manager",
+) -> pd.DataFrame:
+    """Produces assets to impute values for a given timeseries table/column.
+
+    Args:
+        input_asset_name: Name of upstream asset to perform imputation on.
+        output_asset_name: Name of final output asset with imputed column.
+        years: List of years to perform imputation on.
+        value_col: Column imputation will be performed on.
+        imputed_value_col: Name of column in output asset with imputed values.
+        reported_value_col: Name of column in output asset with original reported values.
+        id_col: Name of column identifying entities to group timeseries by.
+        output_io_manager_key: IO-manager to use for final output asset.
+    """
+    timeseries_matrix_asset = f"_{output_asset_name}_timeseries_matrix"
+    cleaned_timeseries_matrix_asset = f"_{output_asset_name}_cleaned_timeseries_matrix"
+    flags_asset = f"_{output_asset_name}_timeseries_matrix_flags"
+
+    @asset(
+        ins={"input_df": AssetIn(input_asset_name)},
+        name=timeseries_matrix_asset,
+    )
+    def _prepare_timeseries_matrix(input_df: pd.DataFrame) -> pd.DataFrame:
+        """Take input timeseries table and convert to timeseries matrix for imputation.
+
+        Returns:
+            Input table as a matrix with a `datetime` row index
+            (e.g. '2006-01-01 00:00:00', ..., '2019-12-31 23:00:00')
+            in local time ignoring daylight-savings,
+            and a `id_col` column index (e.g. 101, ..., 329).
+        """
+        # Convert from datetime_utc to local datetime
+        localized_df = utc_dataframe_to_local(
+            input_df.rename(columns={value_col: "value_col", id_col: "id_col"})
+        )
+        logger.info("Localized timeseries.")
+
+        # Pivot to timeseries matrix
+        matrix = localized_df.pivot(
+            index="datetime", columns="id_col", values="value_col"
+        )
+        return matrix
+
+    @multi_asset(
+        ins={"matrix": AssetIn(timeseries_matrix_asset)},
+        outs={cleaned_timeseries_matrix_asset: AssetOut(), flags_asset: AssetOut()},
+    )
+    def _flag_timeseries_matrix(
```

Comment:
In the DAG I notice that there's now an asset just called `_flag_timeseries_matrix`. If we are using this factory in multiple places for EIA-930 and FERC-714 will that result in a name collision? Does the asset need a use-case specific name specified in the decorator?